package/MDAnalysis/coordinates/DCD.py

```diff
@@ -402,23 +399,22 @@ def _write_next_frame(self, ag):
         :meth:`DCDWriter.write`  takes a more general input
 
 
-        .. deprecated:: 1.0.0
-           Deprecated use of Timestep as argument. To be removed in version 2.0
         .. versionchanged:: 1.0.0
            Added ability to pass AtomGroup or Universe.
            Renamed from `write_next_timestep` to `_write_next_frame`.
+        .. versionchanged:: 2.0.0
+           Deprecated support for Timestep argument has now been removed.
+           Use AtomGroup or Universe as an input instead.
         """
-        if isinstance(ag, base.Timestep):
-            ts = ag
-        else:
+        try:
+            ts = ag.ts
```

Comment:
Constructing an AtomGroup timestep ain’t free, so maybe use ag.positions to grab the positions etc

Reply:
So from a quick look, I think DCD, TRJ, TRR, TRZ, XTC, XYZ, and chemfiles all do something similar. In the interest of not making this PR too hard to follow, would you be ok if I opened an issue for this and dealt with it in a separate PR?